lib/qless/worker/base.rb

```diff
@@ -15,13 +15,16 @@ module Workers
 
     class BaseWorker
       attr_accessor :output, :reserver, :log_level, :interval, :paused,
-                    :options
+                    :options, :sighup_handler
 
       def initialize(reserver, options = {})
         # Our job reserver and options
         @reserver = reserver
         @options = options
 
+        # SIGHUP handler
+        @sighup_handler = options.fetch(:sighup_handler) { lambda { } }
```

Comment:
Huh, that's cool. I didn't know about this. It would be cool to log when workers die and how, but in the situation that they're being SIGHUPped it really doesn't matter. I'd expect that most SIGHUPs are the work of god (the gem), and most SIGKILLs are the works of the OOM Killer. When receiving signals like these, the stack trace can be at any random point in the execution of worker code any way.